router/connection_maker.go

```diff
@@ -48,30 +48,38 @@ func (cm *ConnectionMaker) Start() {
 	go cm.queryLoop(actionChan)
 }
 
-func (cm *ConnectionMaker) InitiateConnection(peer string) error {
-	host, port, err := net.SplitHostPort(peer)
-	if err != nil {
-		host = peer
-		port = "0" // we use that as an indication that "no port was supplied"
-	}
-	addr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("%s:%s", host, port))
-	if err != nil {
-		return err
+func (cm *ConnectionMaker) InitiateConnections(peers []string) error {
+	addrs := make(map[string]*net.TCPAddr)
+	for _, peer := range peers {
+		host, port, err := net.SplitHostPort(peer)
+		if err != nil {
+			host = peer
+			port = "0" // we use that as an indication that "no port was supplied"
+		}
+		addr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("%s:%s", host, port))
+		if err != nil {
+			return err
```

Comment:
If we are trying to later do an atomic replace of a all peers, I'd personally expect it to replace all-or-none of the peers I provided.

Since it only took one peer before, I'm not sure which existing behaviour you're talking about.